src/models/spaces/charts/chart-data.ts

```diff
@@ -4,9 +4,15 @@ import { ChartDataSetModel, ChartDataSetModelType, ChartColors } from "./chart-d
 export const ChartDataModel = types
   .model("ChartData", {
     name: types.string,
-    dataSets: types.array(ChartDataSetModel)
+    dataSets: types.array(ChartDataSetModel),
+    labels: types.array(types.string)
   })
   .views(self => ({
+    get chartLabels() {
+      if (self.labels && self.labels.length > 0) {
+        return self.labels;
+      } else return [];
+    },
```

Comment:
These are the chart level labels as opposed to the labels on the datasets (this is where we store the assayed region for future display in the chart)

Reply:
Oddly, I was originally just returning `return self.labels` but mobx was throwing errors left and right about accessing an undefined array.  This explicit check and return of `[]` fixed it.

Reply:
'data.labels' does the same thing.  We get the following error:
`read an array index (0) that is out of bounds (0). Please check length first. Out of bound indices will not be tracked by MobX`